### ISSUE TITLE
[Core] Add actor name to runtime context

### ISSUE THREAD

**Issue by shiranbi07** [ray-project/ray#27781]
### Description

access actor name (the one given by the client which created the actor) via runtime context

### Use case

I have multiple actors of the same class
I give each one a different name and would like to use that name in the actor logs
The only way to do it today is to pass it to the actor

**Comment by iycheng** [comment]
@wilsonwang371  maybe you can help this one. I think what we need to do is that:

```
@ray.remote
class A:
    def work(self):
        cxt =ray.get_runtime_context()
        cxt.get_actor_name() # this will return the actor name if it's a named actor.
```

**Comment by scottsun94** [comment]
@shiranbi07 could you explain a bit more about your use case?

> I give each one a different name and would like to use that name in the actor logs
> The only way to do it today is to pass it to the actor

What does "use that name in the actor logs" mean? Could you provide an example?
"The only way to do it today is to pass it to the actor" Not sure what this means?

**Comment by shiranbi07** [comment]
@scottsun94 
in my actor constructor i create a log file of my own (per actor instance)
the name of the file should be unique so it can't be class name, i wanted to give it the actor name that I give it when I create the actor on the client side 
for example if i create
counter1 = Counter.options(name="some_name1").remote()
counter2 = Counter.options(name="some_name2").remote()

i want to access some_name1 and some_name2 from within the Counter code (in the __init__ as well as the function calls)
does it make more sense?

**Comment by scottsun94** [comment]
It makes sense. Thanks for the context! @shiranbi07

**Comment by jjyao** [comment]
Implemented by #39347
